ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -100,46 +107,41 @@ function check_history() {
     fi
 }
 
-if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
     export KUBECONFIG=${SHARED_DIR}/kubeconfig
 fi
 
 # Setup proxy if it's present in the shared dir
-if test -f "${SHARED_DIR}/proxy-conf.sh"
-then
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
     # shellcheck disable=SC1091
     source "${SHARED_DIR}/proxy-conf.sh"
 fi
 
-echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
-echo "RELEASE_IMAGE_TARGET is ${RELEASE_IMAGE_TARGET}"
-
-echo -e "Current cluster version: oc get clusterversion\n"
-oc get clusterversion
+get_target_release
+for target in "${TARGET_RELEASES[@]}"
+do
+    echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
+    echo "RELEASE_IMAGE_TARGET is ${RELEASE_IMAGE_TARGET}"
 
-echo -e "RELEASE_IMAGE_LATEST release info:\n"
-oc adm release info "${RELEASE_IMAGE_LATEST}"
+    echo -e "Current client/server version: oc version\n"
+    oc version
 
-echo -e "RELEASE_IMAGE_TARGET release info:\n"
-oc adm release info "${RELEASE_IMAGE_TARGET}"
+    source_minor_version="$(oc get clusterversion --no-headers | awk '{print $2}' | cut -f2 -d.)"
+    echo -e "Source release minor version is: ${source_minor_version}"
 
-target="${RELEASE_IMAGE_TARGET}"
+    target_version="$(oc adm release info "${target}" --output=json | jq -r '.metadata.version')"
+    target_minor_version="$(echo "${target_version}" | cut -f2 -d.)"
+    echo -e "Target release version is: ${target_version}\nTarget minor version is: ${target_minor_version}"
 
-source_minor_version="$(oc get clusterversion --no-headers | awk '{print $2}' | cut -f2 -d.)"
-echo -e "Source release minor version is: ${source_minor_version}"
-
-target_version="$(oc adm release info "${target}" --output=json | jq -r '.metadata.version')"
-target_minor_version="$(echo "${target_version}" | cut -f2 -d.)"
-echo -e "Target release version is: ${target_version}\nTarget minor version is: ${target_minor_version}"
-
-FORCE_UPDATE=false
-if ! check_signed; then
-    echo "You're updating to an unsigned images, you must override the verification using --force flag"
-    FORCE_UPDATE=true
-else
-    admin_ack
-fi
+    FORCE_UPDATE=false
+    if ! check_signed; then
+        echo "You're updating to an unsigned images, you must override the verification using --force flag"
+        FORCE_UPDATE=true
+    else
+        admin_ack
```

Comment:
I think `admin_ack` is not on the same level as `check_signed`, if payload is not signed, allow need `FORCE_UPDATE`,   whatever it is already `admin_ack` or not.

Reply:
How about this, just for easy understanding for the logic.
```
    FORCE_UPDATE=false
    if ! check_signed; then
        echo "You're updating to an unsigned images, you must override the verification using --force flag"
        FORCE_UPDATE=true
    end

    if [[ "${FORCE_UPDATE}" == "false" ]]
        admin_ack
    fi
```

Or even we can move `FORCE_UPDATE` judgement into `admin_ack` function?

Reply:
Thanks for the suggestion! Let me update.